lib/reline/unicode.rb

```diff
@@ -670,6 +670,19 @@ def self.vi_backward_word(line, byte_pointer)
     [byte_size, width]
   end
 
+  def self.common_prefix(list, ignore_case: false)
```

Comment:
Just curious, what's the purpose of this method?

Reply:
Calculates the common prefix of the list
`common_prefix(["a.singleton_class", "a.singleton_method", "a.singleton_methods"])` returns
the common part prefix `"a.singleton_"`
Used in TAB completion
```
$ irb --noautocomplete
irb(main):001> a=''
=> ""
irb(main):002> a.sin[TAB]
↓
irb(main):002> a.singleton_
```

`complete_internal_proc` was doing this common prefix calculation (and also doing other things like filtering list and menu setup)
I split it to `common_prefix` (Pure string processing logic independent to line_editor) and `filter_normalize_candidates` (including completion logic).

Reply:
I understand. Thank you!